### PR TITLE
Remove berkeley db

### PR DESCRIPTION
**Pull request by JesusRami**

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
Looks good, I assume the other two commits are for convenience.

File: src/blockchain_utilities/blockchain_import.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ int check_flush(cryptonote::core &core, std::vector<block_complete_entry> &block
 
     // process block
 
-    block_verification_context bvc = boost::value_initialized<block_verification_context>();
```
Comment (TheCharlatan) [inline src/blockchain_utilities/blockchain_import.cpp]: You're adding this context and then removing it again in this commit. Can you squash this, so you don't touch the lines at all?

File: tests/functional_tests/CMakeLists.txt [hunk 2]
```diff
@@ -51,7 +51,10 @@ target_link_libraries(functional_tests
     ${EXTRA_LIBRARIES})
 
 execute_process(COMMAND ${PYTHON_EXECUTABLE} "-c" "import requests; print('OK')" OUTPUT_VARIABLE REQUESTS_OUTPUT OUTPUT_STRIP_TRAILING_WHITESPACE)
+  message(WARNING "functional_tests_rpc skipped, needs the 'requests' python module")
+
 if (REQUESTS_OUTPUT STREQUAL "OK")
+  message(WARNING "${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_BINARY_DIR} all")
```
Comment (TheCharlatan) [inline tests/functional_tests/CMakeLists.txt]: This seems unrelated. Why are you changing this?

File: src/blockchain_db/blockchain_db.cpp [hunk 3]
```diff
@@ -100,10 +100,10 @@ BlockchainDB *new_db(const std::string& db_type)
...
   if (db_type == "lmdb")
     return new BlockchainLMDB();
-#if defined(BERKELEY_DB)
-  if (db_type == "berkeley")
-    return new BlockchainBDB();
-#endif
+//#if defined(OPTIONAL_DB)
+//  if (db_type == "optional")
+//    return new BlockchainODB();   ///Indicate right type of blockchain
+//#endif
```
Comment (TheCharlatan) [inline src/blockchain_db/blockchain_db.cpp]: I'd rather just remove all the logic that allows for db switching instead of commenting out some pseudo code to indicate how db switching could be achieved.

File: src/blockchain_db/CMakeLists.txt [hunk 4]
```diff
@@ -46,12 +46,12 @@ set(blockchain_db_private_headers
...
-  ${blockchain_db_private_headers}
-  berkeleydb/db_bdb.h
-  )
-endif()
+#if (OPTIONAL_DB)
+#  set(blockchain_db_private_headers
+#  ${blockchain_db_private_headers}
+#  path_to_optional_db.h
+#  )
+#endif()
```
Comment (TheCharlatan) [inline src/blockchain_db/CMakeLists.txt]: Further down, you should also remove the entry for the BDB_LIBRARY

**Comment by moneromooo-monero** [comment]
ping

**Comment by moneromooo-monero** [comment]
whyiamroot rebased (and expanded) this, so I'll close this bug.

+duplicate
